craft_providers/bases/buildd.py

```diff
@@ -301,6 +303,55 @@ def _ensure_instance_config_compatible(
             "Instance is compatible with compatibility tag %r", config.compatibility_tag
         )
 
+    def _ensure_setup_completed(
+        self, *, executor: Executor, deadline: Optional[float]
+    ) -> None:
+        """Ensure the instance was fully setup.
+
+        The last step of setting up an instance is to set the `setup` key in the
+        instance config file to True. This flag is used to verify that setup was
+        completed for the instance.
+
+        :raises BaseConfigurationError: If setup was not completed.
+        """
+        _check_deadline(deadline)
+        brief = "Instance is not properly setup."
+        resolution = "Clean instance and retry the requested operation."
+
+        try:
+            config = InstanceConfiguration.load(
+                executor=executor,
+                config_path=self.instance_config_path,
+            )
+        except ValidationError as error:
+            raise BaseConfigurationError(
+                brief=brief,
+                details="Failed to parse instance configuration file.",
+                resolution=resolution,
+            ) from error
+        except FileNotFoundError as error:
+            raise BaseConfigurationError(
+                brief=brief,
+                details="Failed to find instance config file.",
+                resolution=resolution,
+            ) from error
+
+        if config is None:
+            raise BaseConfigurationError(
+                brief=brief,
+                details="Instance config is empty.",
+                resolution=resolution,
+            )
+
+        if not config.setup or config.setup is False:
```

Comment:
What does this mean for existing instances (pre-`setup` attribute)? Are they invalidated anyway when craft-providers is updated?

Reply:
Excellent question.  Right now, those existing instances would be considered misconfigured and the `BaseConfigurationError` would be raised.

I've been thinking this is an appropriate time to bump the base compatibility tag from `buildd-base-v0` to `buildd-base-v1`.  

This would prevent the existing instances would from being used.  New instances would be created instead.

Reply:
Bumped to `buildd-base-v1`